File: 5.implementation/NT/redsun/src/main/webapp/static/js/services/TeamsService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('teamService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(team) {
			var urlServer = $rootScope.makePostURL("/team/insert");
			return $http.post(urlServer, team);
		}
		
		// Update.
		, update: function(teamId, team) {
			var urlServer = $rootScope.makePostURL("/team/update/" + teamId);
			return $http.put(urlServer, team);
		}
		
		// Delete.
		, delete: function(teamId) {
			var urlServer = $rootScope.makePostURL("/team/delete/" + teamId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(teamId) {
			var url = $rootScope.makeGetURL("/team/getbyid/" + teamId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, team) {
			var url = $rootScope.makeGetURL("/team/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: team });
		}
		
	}// return.
});

File: 5.implementation/redsun-service/src/main/java/com/redsun/service/service/TeamsService.java
package com.redsun.service.service;

import com.redsun.service.entities.Teams;
import com.redsun.service.entities.Result;

/**
 * Teams Service interface
 */
public interface TeamsService {
	
    // Insert.
	Result insert(final Teams teams) ;

    // Update.
	Result update(final Teams teams) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listTeamsForPageAndFilter(final int itemsPerPage, final int pageNo, final Teams teams);

}
